hell/adventcalendar/2024/14/index.md

```diff
@@ -12,147 +12,185 @@ active: true
-
-* To prevent symbols from being redundantly read aloud by screen readers, SVGs and images are given `aria-hidden="true"`. The link functionality is described exclusively via the aria-label in the `<a>` tag (e.g., "Next Page"), ensuring a clear and precise output for screen readers.
-<!-- MM: Why aria-label and not visually hidden text? -->
-<!-- KS: This is my primary question throughout this post. Text content
-     of `<a>` elements is very well understood as an accessible name,
-     and should be preferred over aria- attributes. -->
-* The alt attribute for images remains empty (`alt=""`) to prevent purely visual images from being unnecessarily read aloud. Since the link’s functionality is already described by the `aria-label`, this avoids redundant information for screen reader users.
-<!-- MM: But instead of the aria-label you could use the alt, no? Why is aria-label better? -->
-<!-- MM: Why the extra aria-hidden=true in the code example? -->
-* SVG icons also include the `focusable="false"` attribute to prevent them from being focusable in keyboard navigation. This attribute is unnecessary for images, as they are automatically excluded from keyboard focus.
-<!-- MM: Do you know if there's a browser or a situation where this is still the case? Afaik SVGs only use to be focusable in IE 11. -->
+**Using SVGs**
+- Icons should always have descriptive text to make their function clear for screenreader users. Instead of describing the SVG itself (e.g. “Double arrow pointing left”), the purpose should be communicated (e.g. “First page”).
+- Using a `<span>` element with the [screenreader-only class .sr-only](https://webaim.org/techniques/css/invisiblecontent/) is considered a best practice. This class hides the text visually for sighted users but makes it accessible to screenreaders. One advantage over using an `aria-label` on the `<a>` element is that the text is read by screenreaders with the correct language and pronunciation.
+	- Example: An English text like “First Page” is read correctly with an English voice when using the `.sr-only` class. However, an `aria-label` text might sometimes be read in the screenreader’s language (e.g. with a German voice and pronunciation), which can sound unnatural and difficult to understand.
+	- The key difference is that `.sr-only` fully exposes the text to screenreaders, while `aria-label` provides metadata that may not always be processed optimally.
+- The `role="presentation"` attribute should be added to SVG elements to prevent some screenreaders from announcing them as standalone elements (e.g. “Link, Image, First page”). This ensures they are interpreted as purely decorative and not announced (e.g. “Link, first page”).
+- The `focusable="false"` attribute on SVG elements prevents them from being focusable via keyboard navigation in certain browsers (e.g. Internet Explorer 11). For images, this attribute is not required, because they are automatically excluded from keyboard navigation.
```

Comment:
Do we really still need this? IE11 is dead.

Reply:
yes, I will delete it.